docs/api/ShallowWrapper/instance.md

```diff
@@ -2,20 +2,62 @@
 
 Gets the instance of the component being rendered as the root node passed into `shallow()`.
 
-NOTE: can only be called on a wrapper instance that is also the root instance.
+NOTE: can only be called on a wrapper instance that is also the root instance. With React `16.x`, `instance()` returns `null` for stateless React component/stateless functional components. See example:
 
+#### Returns (React 16.x)
 
+- `ReactComponent`: The stateful React component instance.
+- `null`: If stateless React component was wrapped.
 
-#### Returns
+#### Returns (React 15.x)
 
-`ReactComponent`: The component instance.
+- `ReactComponent`: The component instance.
 
+#### Example
+
+#### Preconditions
+<!-- eslint react/prop-types: 0, react/prefer-stateless-function: 0 -->
+```jsx
+function Stateless() {
+  return <div>Stateless</div>;
+}
+
+class Stateful extends React.Component {
+  render() {
+    return <div>Stateful</div>;
+  }
+}
+```
+#### React 16.x
+```jsx
+test('shallow wrapper instance should not be null', () => {
+  const wrapper = shallow(<Stateless />);
+  const instance = wrapper.instance();
 
+  expect(instance).not.toBe(null); // This will fail, because instance is null
```

Comment:
this seems to be jest matcher syntax; we use chai's expect. could you update it to be `expect(instance).not.to.equal(null)` instead? (same throughout)

Reply:
Will change.